pep/app/modifiers/did-value-update.ts

```diff
@@ -0,0 +1,20 @@
+import Modifier from 'ember-modifier';
+
+/**
+ * Modifier to compare the current value against the previous value. If it has changed, we call the passed in function
+ *
+ * @export
+ * @class DidValueUpdateModifier
+ * @extends {Modifier}
+ */
+export default class DidValueUpdateModifier extends Modifier {
+    previousValue: any = null;
+
+    didReceiveArguments() {
+        if (this.previousValue !== this.args.positional[1] && this.args.positional[0]) {
+            this.previousValue = this.args.positional[1];
+            // eslint-disable-next-line @typescript-eslint/ban-types
+            (this.args.positional[0] as Function)();
+        }
+    }
```

Comment:
Its checking the actual value and comparing it against the previous value. Due to the immutable update, it was getting a new object and running this change even though the values were the same. So I created my own modifier that sorta acts like reacts usePrevious

Reply:
ah interesting, for some reason I thought {{did-update}} did that already, where it would only run if a new value was sent in, not everytime args where passed down